datahub-graphql-core/src/main/resources/gms.graphql

```diff
@@ -1807,3 +1820,159 @@ type Deprecation {
     """
     actor: String
 }
+
+type DataFlow implements Entity {
+    """
+    The unique flow URN
+    """
+    urn: String!
+
+    """
+    GMS Entity Type
+    """
+    type: EntityType!
+
+    """
+    Workflow orchestrator ex: Azkaban, Airflow
+    """
+    orchestrator: String!
+
+    """
+    Id of the flow
+    """
+    flowId: String!
+
+    """
+    Cluster of the flow
+    """
+    cluster: String!
+
+    """
+    Information about a Data processing flow
+    """
+    info: DataFlowInfo
+
+    """
+    Ownership metadata of the flow
+    """
+    ownership: Ownership
+}
+
+type DataFlowInfo {
+    """
+    Name of the flow
+    """
+    name: String!
+
+    """
+    Description of the flow
+    """
+    description: String
+
+    """
+    Optional project/namespace associated with the flow
+    """
+    project: String
+}
+
+
+type DataJob implements Entity {
+    """
+    The unique job URN
+    """
+    urn: String!
+
+    """
+    GMS Entity Type
+    """
+    type: EntityType!
+
+    """
+    The associated data flow
+    """
+    dataFlow: DataFlow!
+
+    """
+    Id of the job
+    """
+    jobId: String!
+
+    """
+    Ownership metadata of the job
+    """
+    ownership: Ownership
+
+    """
+    Information about the inputs and outputs of a Data processing job
+    """
+    inputOutput: DataJobInputOutput
+
+    """
+    Information about a Data processing job
+    """
+    info: DataJobInfo
+}
+
+type DataJobInfo {
+    """
+    Job name
+    """
+    name: String!
+
+    """
+    Job description
+    """
+    description: String
+
+    """
+    Datajob type
+    """
+    type: AzkabanJobType
+}
+
+type DataJobInputOutput {
+    """
+    Input datasets produced by the data job during processing
+    """
+    inputDatasets: [Dataset!]
+
+    """
+    Output datasets produced by the data job during processing
+    """
+    outputDatasets: [Dataset!]
+}
+
+enum AzkabanJobType {
```

Comment:
What if we just made this "JobType" and tried to generify it further? Alternatively we could have a union of complex objects, one per type.